src/nix/repl.cc

```diff
@@ -671,13 +673,15 @@ void NixRepl::reloadFiles()
     Strings old = loadedFiles;
     loadedFiles.clear();
 
-    bool first = true;
     for (auto & i : old) {
-        if (!first) notice("");
-        first = false;
         notice("Loading '%1%'...", i);
         loadFile(i);
     }
+
+    for (auto & [i,what] : getValues()) {
+        notice("Loading Installable '%1%'...", what);
+        addAttrsToScope(*i);
```

Comment:
So this effectively merges the attributes of multiple installables?

I don't like that "loadables" are loaded into the global scope anyway. It makes discovering which attributes exist harder than it should.

Perhaps flakes and installables could be put in a binding that's derived from the installable name?

```
# nix repl nixpkgs
Loading Installable 'flake:nixpkgs#'...
Added 1 variable.

nix-repl> nixpkgs.legacyPackages.x86_64-linux.emacs.name
"emacs-27.1"

nix-repl> nixpkgs.<TAB>
<UX delight ensues as all top-level flake attrs are listed>

```

Also interesting

```
# nix repl . .#inputs.nixpkgs
Added 2 variables

nix-repl> foo.<TAB>

nix-repl> nixpkgs.lib.concatStringsSep ":" ["testing" "it"]

```

Reply:
The only problem with this is that flakes don't have an obvious name, so it's not entirely straightforward what the top-level attribute name (e.g. `nixpkgs`) should be.

Reply:
Right. I pulled `foo` out of thin air.
Most flake refs have a somewhat straightforward candidate for a name.
Using `dirname $PWD` for `.` has produced mixed success in the past though.

Maybe the attribute name should be provided by the user.

```
nix repl --bind foo . --bind nixpkgs .#inputs.nixpkgs
```

or

```
nix repl foo=. nixpkgs=.#inputs.nixpkgs
```

Adding a bare flake or installable to the top-level scope seems ok-ish as long as there's only one. Or maybe it should default to `flake=` for a flake and a name derived from the installable otherwise?

Reply:
We do have the previous behavior of multiple `--files` to worry about.

I wouldn't mind if it just made an array called `args`, hah! Then don't have to worry about names, and `:a` can put whatever one likes at the top level.